include/deal.II/numerics/matrix_tools.h

```diff
@@ -232,23 +234,23 @@ namespace MatrixCreator
    *
    * See the general documentation of this namespace for more information.
    */
-  template <int dim, typename number, int spacedim>
+  template <int dim, int spacedim, typename number>
   void create_mass_matrix (const Mapping<dim, spacedim>       &mapping,
                            const DoFHandler<dim,spacedim>    &dof,
                            const Quadrature<dim>    &q,
                            SparseMatrix<number>     &matrix,
-                           const Function<spacedim> *const a = 0,
+                           const Function<spacedim,number> *const a = 0,
```

Comment:
Here and in many of the functions below, I don't think it's actually necessary that the `Function` argument uses the same scalar type as what we use in the matrix and vector arguments to the function. It's certainly not wrong, and I don't in principle object to it, but it *is* an incompatible change to require this. I just wanted to put it up for discussion.